metrics/protocol/shannon/metrics.go

```diff
@@ -741,6 +780,20 @@ func processRelayMinerErrors(
 	}
 }
 
+// SetActiveHTTPRelays updates the gauge metric with the current number of active HTTP relays.
+// This should be called whenever the active HTTP relay count changes in the concurrency limiter.
+func SetActiveHTTPRelays(activeCount int64) {
```

Comment:
This needs a TODO_TECHDEBT: the metrics package should use the passed observation to report metrics: it should not expose any methods for external usage (other than PublishMetrics)